src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -220,13 +220,26 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
     }
 
     private fun checkDuplicateImport(holder: RsAnnotationHolder, useSpeck: RsUseSpeck) {
+        if (checkDuplicateSelfInUseGroup(holder, useSpeck)) return
+
         val duplicates = holder.currentAnnotationSession.duplicatesByNamespace(useSpeck.containingMod, false)
         if (useSpeck.namespaces.any { useSpeck in duplicates[it].orEmpty() }) {
             val identifier = PsiTreeUtil.getDeepestLast(useSpeck)
             RsDiagnostic.DuplicateImportError(identifier).addToHolder(holder)
         }
     }
 
+    private fun checkDuplicateSelfInUseGroup(holder: RsAnnotationHolder, useSpeck: RsUseSpeck): Boolean {
+        val useGroup = useSpeck.parent as? RsUseGroup ?: return false
+        val selfImports = useGroup.useSpeckList.filter { it.text == "self" }
```

Comment:
Maybe `path.referenceName` is more suitable than just `text`

Reply:
Looks like we could calculate `selfImports` once per `RsUseGroup` and then report problems about each extra `self`. In contrast, currently, we recalculate `selfImports` when processing each `RsUseSpeck`. I am not sure if this could cause performance problems because very big `RsUseGroup` are rare I guess, but anyway, we should consider this potential problem. @dima74 WDYT?

Reply:
Sounds good!